services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -1967,6 +1975,47 @@ specPatchUser = do
       let patchOp = PatchOp.PatchOp [removeAttrib "externalId"]
       patchUser_ (Just tok) (Just userid) patchOp (env ^. teSpar) !!! const 400 === statusCode
 
+testPatchRole :: (Text -> [Role] -> Operation) -> TestSpar ()
+testPatchRole replaceOrAdd = do
+  env <- ask
+  let brig = env ^. teBrig
+  let galley = env ^. teGalley
+  (owner, tid) <- call $ createUserWithTeam brig galley
+  tok <- registerScimToken tid Nothing
+  let testWithInitialRole r = forM_ [minBound .. maxBound] (testPatchRoles brig replaceOrAdd tid owner tok r)
+  forM_ [minBound .. maxBound] testWithInitialRole
```

Comment:
i suggested a different way to write this, but this time i have another suggestion :-)

```suggestion
  let values = [[], ["member", "admin"], ["notarole"]] <> (:[]) <$> [minBound .. maxBound]
      testWithInitialRole r = forM_ values (testPatchRoles brig replaceOrAdd tid owner tok r)
  forM_ values testWithInitialRole
```

(this would require to take `[Role]` instead of `Role` in initial and target below.)

Reply:
maybe this trick could also be used to simplify the tests for `POST`, `PUT`?

Reply:
> i suggested a different way to write this, but this time i have another suggestion :-)
I know, but that didn't compile, and I tried to come as close as possible.